main.go

```diff
@@ -54,29 +55,100 @@ func main() {
-
 	log.SetEnableDebugLog(config.DebugMode)
 
 	absDeployPth, err := pathutil.AbsPath(config.DeployPath)
 	if err != nil {
 		fail("Failed to expand path: %s, error: %s", config.DeployPath, err)
 	}
 
-	filesToDeploy := []string{}
-
 	tmpDir, err := pathutil.NormalizedOSTempDirPath("__deploy-to-bitrise-io__")
 	if err != nil {
 		fail("Failed to create tmp dir, error: %s", err)
 	}
 
-	// Collect files to deploy
+	filesToDeploy := collectFilesToDeploy(absDeployPth, config, tmpDir)
+	clearedFilesToDeploy := clearDeployFiles(filesToDeploy)
+	logDeployFiles(clearedFilesToDeploy)
+
+	fmt.Println()
+	log.Infof("Deploying files")
+
+	publicInstallPages := deploy(clearedFilesToDeploy, config)
+	fmt.Println()
+	log.Donef("Success")
+	log.Printf("You can find the Artifact on Bitrise, on the Build's page: %s", config.BuildURL)
+
+	exportInstallPages(publicInstallPages, config)
+	deployTestResults(config)
+}
+
+func exportInstallPages(publicInstallPages map[string]string, config Config) {
+	if len(publicInstallPages) > 0 {
+		temp := template.New("Public Install Page template")
+		var pages []PublicInstallPage
+		for file, url := range publicInstallPages {
+			pages = append(pages, PublicInstallPage{
+				File: file,
+				URL:  url,
+			})
+		}
+
+		if err := tools.ExportEnvironmentWithEnvman("BITRISE_PUBLIC_INSTALL_PAGE_URL", pages[0].URL); err != nil {
+			fail("Failed to export BITRISE_PUBLIC_INSTALL_PAGE_URL, error: %s", err)
+		}
+		log.Printf("The public install page url is now available in the Environment Variable: BITRISE_PUBLIC_INSTALL_PAGE_URL (value: %s)\n", pages[0].URL)
+
+		temp, err := temp.Parse(config.PublicInstallPageMapFormat)
+		if err != nil {
+			fail("Error during parsing PublicInstallPageMap: ", err)
+		}
+
+		buf := new(bytes.Buffer)
+		if err := temp.Execute(buf, pages); err != nil {
+			fail("Execute: ", err)
+		}
+
+		if err := tools.ExportEnvironmentWithEnvman("BITRISE_PUBLIC_INSTALL_PAGE_URL_MAP", buf.String()); err != nil {
+			fail("Failed to export BITRISE_PUBLIC_INSTALL_PAGE_URL_MAP, error: %s", err)
+		}
+		log.Printf("A map of deployed files and their public install page urls is now available in the Environment Variable: BITRISE_PUBLIC_INSTALL_PAGE_URL_MAP (value: %s)", buf.String())
+		log.Printf("")
+	}
+}
+
+func logDeployFiles(clearedFilesToDeploy []string) {
+	fmt.Println()
+	log.Infof("List of files to deploy")
+	for _, pth := range clearedFilesToDeploy {
+		log.Printf("- %s", pth)
+	}
+}
+
+func clearDeployFiles(filesToDeploy []string) []string {
+	clearedFilesToDeploy := []string{}
+	for _, pth := range filesToDeploy {
+		for _, fileBaseNameToSkip := range fileBaseNamesToSkip {
+			if filepath.Base(pth) == fileBaseNameToSkip {
+				log.Warnf("skipping: %s", pth)
+			} else {
+				clearedFilesToDeploy = append(clearedFilesToDeploy, pth)
+			}
+
+		}
+	}
+	return clearedFilesToDeploy
+}
+
+func collectFilesToDeploy(absDeployPth string, config Config, tmpDir string) []string {
+	filesToDeploy := []string{}
```

Comment:
you can create this var if you use named return

Reply:
Changed to named return.